### ISSUE TITLE
Easy-Search filtering through tags, not working.

### ISSUE THREAD

**Issue by mcostasoftware** [matteodem/meteor-easy-search#216]
I'm coding a web application where users can create posts which are characterized through the use of tags. When searching for posts a user can add a tag to filter the search.

I'm using the code found at http://matteodem.github.io/meteor-easy-search/docs/recipes/ with some adaptations.

When a new tag is added an event is triggered and the following code is ran.
(/client/views/templates/sidebar.js)
```javascript
            var instance = EasySearch.getComponentInstance(
                    { index : 'suggestions' }
                );
            EasySearch.changeProperty('suggestions', 'tags', tagsL);
            instance.triggerSearch();
```
In another file I have written this code:
(/lib/collections/suggestions.js)

```javascript
Suggestions.initEasySearch('name');

EasySearch.createSearchIndex('suggestions', {
    'field' : ['name'],
    'collection' : Suggestions,
    'props' : {
        'tags' : []
    },
    'query' : function (searchString) {
        // Default query that will be used for searching
        var query = EasySearch.getSearcher(this.use).defaultQuery(this, searchString);

        // filter for categories if set
        if (this.props.tags.length > 0) {
            query.tags =  {$in: this.props.tags};
        }

        return query;
    }
});
```
The code simply isn't working. I'm not getting any errors but the search results are exactly the same as before adding the code shown above.

Also, it might be important to know what a post (suggestion) looks like.
```javascript
{
            name: title,
            score: 0,
            description: description,
            tags : tags,
            createdBy: Meteor.userId(),
            createdAt: new Date(),
            username: Meteor.user().username
        }
```

**Comment by matteodem** [comment]
What I see is that in following line:

```javascript
EasySearch.changeProperty('suggestions', 'Tags', x);
```

should be following (tags all lowercase):

```javascript
EasySearch.changeProperty('suggestions', 'tags', x);
```

Also is this code inside an event function? And x should be an array.

**Comment by mcostasoftware** [comment]
@matteodem I'm narrowed down the problem and fixed the problems that you pointed out.

Everything's working except {$in: this.props.tags} doesn't do what I want.

I've been reading meteor's documentation on Mongo-Style Selectors to try and use ElemMatch which is appropriate for what I want.

```javascript
 query.tags =  {$elemMatch: { 'tags': this.props.tags} } ;
```
I tried the code above but it doesn't work (I can see why it doesn't work but I don't know how to fix it).
Basically, I want to be able to use $elemMatch.

I tried the code below (from the recipes) but it doesn't seem to work.

```javascript
query.$or.push({
                suggestions: {
                    $elemMatch: {
                        tags: this.props.tags
                    }
                }
            });
```
